src/game-engine/world/actor/actor.ts

```diff
@@ -69,6 +72,63 @@ export abstract class Actor {
         return remainingHitpoints === 0 ? 'dead' : 'alive';
     }
 
+    /**
+     * Waits for the actor to reach the specified position before resolving it's promise.
+     * The promise will be rejected if the actor's walking queue changes or their movement is otherwise canceled.
+     * @param position The position that the actor needs to reach for the promise to resolve.
+     */
+    public async waitForPathing(position: Position): Promise<void>;
+
+    /**
+     * Waits for the actor to reach the specified game object before resolving it's promise.
+     * The promise will be rejected if the actor's walking queue changes or their movement is otherwise canceled.
+     * @param gameObject The game object to wait for the actor to reach.
+     */
+    public async waitForPathing(gameObject: LocationObject): Promise<void>;
+
+    /**
+     * Waits for the actor to reach the specified game object before resolving it's promise.
+     * The promise will be rejected if the actor's walking queue changes or their movement is otherwise canceled.
+     * @param target The position or game object that the actor needs to reach for the promise to resolve.
+     */
+    public async waitForPathing(target: Position | LocationObject): Promise<void>;
```

Comment:
Why is this line duplicated? @Tynarus

Reply:
This is how overloads function in typescript - unfortunately for some odd reason you have to duplicate the main method if you want it to be part of the public API when you make an overload something.

To get specific with this particular piece of code, without adding the duplicated line, you would only get typeahead options for...

```typescript
waitForPathing(Position)
waitForPatching(LocationObject)
```

Oddly enough, the default method that is being overloaded doesn't show up automatically. The duplicated line makes it appear and passes the compiler:

```typescript
waitForPathing(Position)
waitForPatching(LocationObject)
waitForPatching(Position | LocationObject)
```

The only instance in which this actually is useful though, is when you're calling that method and passing in a variable as the single parameter, and the variable has a dual type of `Position | LocationObject` instead of just passing in a strictly typed `Position` or `LocationObject`. 

Aaaaand of course I needed that particular use case in the new action priority system. 😅

Reply:
Aaaah ok, thanks for the explanation!